### PR TITLE
feat: ElasticSearch / Kibana multi tenancy and opendistro

### PR DESCRIPTION
**Pull request by shawnsarwar**
This patch adds multi-tenant elasticsearch and Kibana to the cluster.

Once the base initialization is done do:
 - elasticsearch/setup.py
 - elasticsearch/start.py

You can then access Kibana for a realm at 
http://aether.local/{realm}/kibana/kibana-app

The reasoning for the url convention is explained here: https://github.com/eHealthAfrica/gateway-manager/pull/11

### CODE REVIEW

File: elasticsearch/docker-compose.yml [hunk 1]
```diff
@@ -23,17 +23,17 @@ services:
...
-      ELASTICSEARCH_INSTANCE_URLS: "elasticsearch"
-      BOOTSTRAP_SERVERS: "kafka:29092"
-      CONSUMER_PORT: 9099
-      ES_CONSUMER_CONFIG_PATH: "/code/conf/consumer/consumer.json"
-      ES_CONSUMER_KAFKA_CONFIG_PATH: "/code/conf/consumer/kafka.json"
-    links:
-      - elasticsearch
-    networks:
-      - aether
+  # elasticsearch-consumer:
```
Comment (adamvert) [inline elasticsearch/docker-compose.yml]: This block will be uncommented once you finish MTing the Consumer, right?

File: elasticsearch/setup_es.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
```
Comment (adamvert) [inline elasticsearch/setup_es.sh]: Why do we need both `setup.sh` and `setup_es.sh`?

File: docker-compose-base.yml [hunk 3]
```diff
@@ -398,6 +398,10 @@ services:
       ZOOKEEPER_PW: ${ZK_ROOT_PW}
       KAFKA_SECRET: ${KAFKA_SECRET}
 
+      ELASTICSEARCH_HOST: http://elasticsearch:9200
+      ELASTICSEARCH_USER: admin
+      ELASTICSEARCH_PW: ${ELASTIC_PASSWORD}
```
Comment (adamvert) [inline docker-compose-base.yml]: Can we try and be consistent on naming of env vars? Some examples:
- `ES_CONSUMER_CONFIG_PATH`
- `ELASTICSEARCH_HOST`
- `ELASTICSEARCH_PW`
- `ELASTIC_PASSWORD`

Is it possible to pick a prefix format and stick with it (ELASTIC/ELASTICSEARCH/ES), likewise whether to call it PASSOWRD or PW?

File: elasticsearch/docker-compose.yml (hunk 1 above)
Comment (shawnsarwar) [reply elasticsearch/docker-compose.yml]: Yessir

File: elasticsearch/setup_es.sh (hunk 2 above)
Comment (shawnsarwar) [reply elasticsearch/setup_es.sh]: :+1:

File: docker-compose-base.yml (hunk 3 above)
Comment (shawnsarwar) [reply docker-compose-base.yml]: :+1:

File: elasticsearch/docker-compose.yml (hunk 1 above)
Comment (shawnsarwar) [reply elasticsearch/docker-compose.yml]: Yessir

**Comment by shawnsarwar** [comment]
@adamvert I normalized the ENVs to the extent possible. There are still quite a few _PWs in docker-compose files that the containers are expecting. We'll have to make those changes over the next few weeks as we update them.

**Pull request merged**